### PR TITLE
Handle short form queries

### LINKED ISSUE #2855: Support deserialization of query short form
Relates #2836 

Deserialization in NEST should handle the short form of queries such that it is able to correctly deserialize queries that have been constructed outside of NEST.

### PR DESCRIPTION
**Pull request by russcam**
This commit handles the short form of the following queries:

- term query
- match query
- prefix query
- match phrase query
- match phrase prefix query

Closes #2855

### CODE REVIEW

**Comment by russcam** [comment]
This should also be ported to 7.x and master. Introducing an interface for full-text queries with a `Query` property may be a good idea.

File: src/Tests/Tests/QueryDsl/TermLevel/Prefix/PrefixQueryUsageTests.cs [hunk 1]
```diff
@@ -47,5 +52,16 @@ public class PrefixQueryUsageTests : QueryDslUsageTestsBase
 				.Value("proj")
 				.Rewrite(MultiTermQueryRewrite.TopTerms(10))
 			);
+
+		//hide
+		[U] public void DeserializeShortForm()
+		{
+			var json = JsonConvert.SerializeObject(new { description = "project description" });
```
Comment (Mpdreamz) [inline src/Tests/Tests/QueryDsl/TermLevel/Prefix/PrefixQueryUsageTests.cs]: Can we move this to `QueryDslUsageTestsBase` and reuse in the other `DeserializeShortForm` ?

File: src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryJsonConverter.cs [hunk 2]
```diff
@@ -8,14 +8,39 @@ internal class FieldNameQueryJsonConverter<TReadAs> : ReserializeJsonConverter<T
...
+			{
+				case JsonToken.StartObject:
+					query = ReadAs(reader, objectType, existingValue, serializer);
+					break;
+				case JsonToken.Null:
+					break;
+				default:
+					query = new TReadAs();
+					switch (query)
+					{
```
Comment (Mpdreamz) [inline src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryJsonConverter.cs]: We need to cover all of these: 

https://github.com/elastic/clients-flight-recorder/blob/master/script/lib/normalizer.js#L430-L434

Comment (russcam) [reply src/Nest/QueryDsl/Abstractions/FieldName/FieldNameQueryJsonConverter.cs]:
```json
  const fieldQueries = [
    'range', 'common', 'intervals', 'match', 'match_bool_prefix', 'match_phrase', 'match_phrase_prefix',
    'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature',
    'term', 'terms'
  ]
```

I don't think the other cases are needed. It looks like 'range', 'common', 'intervals', 'geo_bounding_box', 'geo_distance', 'geo_polygon', 'geo_shape', 'shape', 'distance_feature', 'rank_feature' and 'terms' query do not support a query short form. The `switch` statement on type of query only applies when the value assigned to the field name key is not a JSON object.

File: src/Tests/Tests/QueryDsl/TermLevel/Prefix/PrefixQueryUsageTests.cs (hunk 1 above)
Comment (russcam) [reply src/Tests/Tests/QueryDsl/TermLevel/Prefix/PrefixQueryUsageTests.cs]: addressed in 306aff9e3bc5418aad7f058ab1e9a1469530b9e0

**Comment by russcam** [comment]
Would you mind taking another look please, @Mpdreamz ?

**Comment by russcam** [comment]
Spoke with @Mpdreamz about this offline, so merging in.

**Pull request merged**